crnk-core/src/main/java/io/crnk/core/engine/internal/information/resource/DefaultResourceInformationProvider.java

```diff
@@ -82,9 +83,21 @@ public ResourceInformation build(Class<?> resourceClass, boolean allowNonResourc
 		String superResourceType = superclass != Object.class && context.accept(superclass) ? context.getResourceType(superclass) : null;
 
 		Class<? extends PagingBehavior> pagingBehaviorType = ClassUtils.getAnnotation(resourceClass, JsonApiResource.class).get().pagingBehavior();
-		java.util.Optional<PagingBehavior> optPagingBehavior = pagingBehaviors.values().stream().filter(it -> pagingBehaviorType.isInstance(it)).findFirst();
-		if (!optPagingBehavior.isPresent()) {
-			throw new IllegalStateException("paging behavior not found: " + pagingBehaviorType);
+
+		// cross check if desired resource paging behavior is a registered behavior, error out if not.
+		// if no behavior is set for the resource, we pick the first from registered behaviors.
+		java.util.Optional<PagingBehavior> optPagingBehavior;
+		if (!pagingBehaviorType.equals(VoidPagingBehavior.class)) {
+			optPagingBehavior = pagingBehaviors.values().stream()
+							.filter(it -> pagingBehaviorType.isInstance(it))
+							.findFirst();
+
+			if (!optPagingBehavior.isPresent()) {
+				throw new IllegalStateException("paging behavior not found: " + pagingBehaviorType);
+			}
+
+		} else {
+			optPagingBehavior = pagingBehaviors.values().stream().findFirst();
 		}
 
```

Comment:
Those lines above would pretty much be the same, even if a delegating `DefaultPagingBehavior` was used (except that I'd have to instantiate a `DefaultPagingBehavior` that delegates to the actual implementation).
I decided against that, as it only makes it a little harder to figure out which behavior is used. @remmeier